.changelog/unreleased/breaking-changes/1041-reconsider-representation-of-keys.md

```diff
@@ -0,0 +1,9 @@
+- Change `store`, `state`, `evidence` and `light` databases key format. The
+  keys are now grouped by height and lexicographically sorted using the
+  [orderedcode](https://github.com/google/orderedcode) library.
+
+  To upgrade, run `cometbft migrate-db`. **Before doing so, it's highly
```

Comment:
saw this pr on my feed, the title intrigued me. This step would prevent the sdk from adopting this change. This will cause downtime when users upgrade, there should be a way to make it lazy so users dont experience this downtime

Reply:
Would it be fine to live migrate the latest {N} blocks which hopefully happened quickly, and then migrate the rest in the background?

Reply:
im not sure there needs to be a migration at the point of upgrade, the node could be able to tell what is the old format vs new, then migrate in the background. Potentially causing 0 donwtime